spacy_llm/ty.py

```diff
@@ -84,7 +93,34 @@ def __call__(self, examples: Iterable[Example], **kwargs) -> Dict[str, Any]:
 
 
 @runtime_checkable
-class LLMTask(Protocol):
```

Comment:
Can we keep the definition of `LLMTask` here as the union between `ShardingLLMTask` and `NonshardingLLMTask`? That would avoid some edits in the test suite - the tasks would still be instances of `LLMTask`, as before.